pragma-sdk/pragma_sdk/common/fetchers/generic_fetchers/deribit.py

```diff
@@ -0,0 +1,304 @@
+            else None,
+            open_interest=float(data["open_interest"]),
+            ask_price=float(data["ask_price"])
+            if data.get("ask_price") is not None
+            else None,
+            bid_price=float(data["bid_price"])
+            if data.get("bid_price") is not None
+            else None,
+            instrument_name=str(data["instrument_name"]),
+            mark_price=float(data["mark_price"]),
+            last=float(data["last"]) if data.get("last") is not None else None,
+            low=float(data["low"]) if data.get("low") is not None else None,
+            high=float(data["high"]) if data.get("high") is not None else None,
+        )
+
+    def extract_strike_price_and_option_type(self) -> Tuple[float, str]:
+        """
+        Retrieve the strike price and the option type from the instrument name.
+        """
+        separate_string = self.instrument_name.split("-")
+        strike_price = float(separate_string[2])
+        option_type = separate_string[3]
+        return (strike_price, option_type)
+
+
+@dataclass
+class OptionData:
+    instrument_name: str
+    base_currency: str
+    current_timestamp: UnixTimestamp
+    mark_price: float
+
+    @classmethod
+    def from_deribit_response(cls, response: DeribitOptionResponse) -> "OptionData":
+        current_timestamp = int(time.time())
+        return cls(
+            instrument_name=response.instrument_name,
+            base_currency=response.base_currency,
+            current_timestamp=current_timestamp,
+            mark_price=response.mark_price * response.underlying_price,
+        )
+
+    def as_dict(self) -> dict:
+        return {
+            "instrument_name": self.instrument_name,
+            "base_currency": self.base_currency,
+            "current_timestamp": self.current_timestamp,
+            "mark_price": self.mark_price,
+        }
+
+    def __hash__(self) -> int:
+        """Computes the Pedersen hash of the OptionData."""
+        instrument_name_felt = str_to_felt(self.instrument_name)
+        base_currency_felt = str_to_felt(self.base_currency)
+        return compute_hash_on_elements(  # type: ignore[no-any-return]
+            [
+                instrument_name_felt,
+                base_currency_felt,
+                self.current_timestamp,
+                int(self.mark_price),
+            ]
+        )
+
+
+CurrenciesOptions = Dict[str, List[OptionData]]
+
+
+@dataclass
+class LatestData:
+    merkle_tree: MerkleTree
+    options: CurrenciesOptions
+
+
+class DeribitOptionsFetcher(FetcherInterfaceT):
+    """
+    Deribit fetcher.
+    Retrieves all the options data for all available instruments for a set of pairs.
+
+    All the available prices are put in a Merkle tree which is sent in a GenericEntry.
+    """
+
+    publisher: str
+    pairs: List[Pair]
+    headers: Dict[Any, Any]
+    _client: PragmaOnChainClient
+    _latest_data: Optional[LatestData] = None
+
+    REQUIRED_PAIRS = {
```

Comment:
we should not need to update the SDK when we just want to add a new option asset tho let's not restrain the pairs at init. It will just throw an error later one if not supported anyways.

Reply:
done